lib/site_prism/page.rb

```diff
@@ -72,11 +75,11 @@ def self.set_url_matcher(page_url_matcher)
     end
 
     def self.url
-      @url
+      @url ||= nil
     end
 
     def self.url_matcher
-      @url_matcher || url
+      @url_matcher ||= nil || url
```

Comment:
First, the nil here will never be used since it will always hit the `||` in this case.

Second, I think it would be better to just initialize `@url_matcher` to `nil` since it is possible if the url_matcher is accessed before the url is set, then this could end up getting memoized as nil and then will never fall back to the `url`.